packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1231,8 +1243,20 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   /// The unique identifier for this node.
   ///
   /// The root node has an id of zero. Other nodes are given a unique id when
-  /// they are created.
-  final int id;
+  /// they are attached.
+  ///
+  /// null if this node is detached.
```

Comment:
Alternative: node id is instantiated when node is created, but on attachment in the overflow case it checks whether the ID is still available and switches then if not. That way, it's never null (but only meaningful when attached).

Reply:
It may also increase the change that a temporarily un-attached node can re-attach under its old ID.